x-pack/legacy/plugins/ingest/server/libs/datasources.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { FrameworkUser } from './adapters/framework/adapter_types';
+import { Datasource } from './adapters/policy/adapter_types';
+
+export class DatasourcesLib {
+  private datasources: Datasource[] = [];
+
```

Comment:
I believe all the `return new Promise` instances in this PR can be removed since a) `async` functions already return a `Promise` b) the current examples are all synchronous operations.